staging/src/k8s.io/cri-api/pkg/apis/runtime/v1/api.proto

```diff
@@ -1512,3 +1517,70 @@ message ReopenContainerLogRequest {
 
 message ReopenContainerLogResponse{
 }
+
+// Common options used for checkpointing and restoring.
+message CheckpointRestoreOptions {
+    // Keep temporary files. Like log files. Helpful for debugging.
+    bool keep = 1;
+    // Checkpoint/Restore the container with established TCP connections.
+    bool tcp_established = 2;
+    // Location of the checkpoint archive used for export/import
+    string archive = 3;
+    // Specify the checkpoint archive compression algorithm
+    // See github.com/containers/storage/pkg/archive/archive.go for
+    // possible values.
+    int64 compression = 4;
+}
+
+message CheckpointContainerOptions {
+    CheckpointRestoreOptions common_options = 1;
+    // Leave the container running after checkpointing
+    bool leave_running = 2;
+}
+
+message CheckpointContainerRequest {
+    // ID of the container or pod to be checkpointed.
+    string id = 1;
+    // Options for checkpointing.
+    CheckpointContainerOptions options = 2;
+}
+
+message CheckpointContainerResponse {
+}
+
+message RestoreContainerOptions {
```

Comment:
I think we could probably leave Restore out for now.

Reply:
Thanks for the review. I think we already briefly discussed this in the KEP and it is mentioned there also. My argument for keeping the `Restore` part in the CRI API it is necessary to restore checkpoints in CRI implementations like CRI-O via crictl.

https://github.com/kubernetes/enhancements/blob/master/keps/sig-node/2008-forensic-container-checkpointing/README.md?plain=1#L111

Reply:
> do you have any proposal(s) for new crictl commands

https://github.com/kubernetes-sigs/cri-tools/pull/662

> I may be way off base.. what is the expected state of the restored container?

Exactly what it was during checkpointing.  What do you mean by state?

Reply:
for forensics I expected the concept of container state to be one of https://github.com/opencontainers/runc/blob/51e607f2cdd8ccf7f39f29aab531d674e487b311/libcontainer/container.go#L29-L42

The docs for runc checkpoint mention .. checkpoint for a "running container"  Does that mean this checkpointing feature will only work if the container is in a running state?

Reply:
in my head there is a run container state matrix for checkpoint and another for restore.. and I don't know how to fill it out exactly.. Apologies, in this review I have not yet tried to build the parts and run it locally.